### PR TITLE
Wal 9587 tf runner pod template

### PR DESCRIPTION
**Pull request by delca85**
This adds another example of using the Terraform Driver with an external runner along with the usage of `runner_pod_template` field in the TF Resource Definition.

Related to https://github.com/humanitec/developers-docs/pull/380.

### CODE REVIEW

File: resource-definitions/terraform-driver/runner-pod-configuration/k8s-cluster-refs.tf [hunk 1]
```diff
@@ -0,0 +1,30 @@
+resource "humanitec_resource_definition" "aks_resource_cluster" {
+  id          = "eks-enabled-cluster"
```
Comment (TobiasBabin) [inline resource-definitions/terraform-driver/runner-pod-configuration/k8s-cluster-refs.tf:2]:
```suggestion
  id          = "aad-enabled-cluster"
```
Do we mean "AAD enabled"? Might be a copy&paste effect.

File: resource-definitions/terraform-driver/runner-pod-configuration/k8s-cluster-refs.tf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  name        = "eks-enabled-cluster"
```
Comment (TobiasBabin) [inline resource-definitions/terraform-driver/runner-pod-configuration/k8s-cluster-refs.tf:3]:
```suggestion
  id          = "aad-enabled-cluster"
```
Same as previous line.

File: resource-definitions/terraform-driver/runner-pod-configuration/k8s-cluster-refs.tf [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      resource_group  = "my-azure-resource-group"
+      subscription_id = "123456-1234-1234-1234-123456789"
+      server_app_id   = "6dae42f8-4368-4678-94ff-3960e28e3630"
+    }
+  }
+}
+
+
+resource "humanitec_resource_definition_criteria" "aks_aad_resource_cluster" {
+  resource_definition_id = humanitec_resource_definition.aks_aad_resource_cluster.id
```
Comment (TobiasBabin) [inline resource-definitions/terraform-driver/runner-pod-configuration/k8s-cluster-refs.tf:28]:
```suggestion
  resource_definition_id = humanitec_resource_definition.aks_resource_cluster.id
```
Align resource name with the humanitec_resource_definition object above. Alternatively, rename the cluster "humanitec_resource_definition" to "aks_aad_resource_cluster".

File: resource-definitions/terraform-driver/runner-pod-configuration/k8s-cluster-refs.tf [hunk 4]
```diff
@@ -0,0 +1,30 @@
+resource "humanitec_resource_definition" "aks_resource_cluster" {
```
Comment (TobiasBabin) [inline resource-definitions/terraform-driver/runner-pod-configuration/k8s-cluster-refs.tf:1]:
```suggestion
resource "humanitec_resource_definition" "aks_aad_resource_cluster" {
```
Align the resource name with the reference in the matching criteria below. Either change this name here to `aks_aad_resource_cluster`, or the reference in the matching criteria to `aks_resource_cluster`.

**Comment by delca85** [comment]
@TobiasBabin I think I fixed the bits you highlighted.

**Pull request merged**